planning/src/parsing/sexpr.rs

```diff
@@ -339,7 +351,27 @@ fn read(tokens: &mut std::iter::Peekable<core::slice::Iter<Token>>, src: &std::s
                 }
             }
         }
+
         Some(Token::RParen(_)) => bail!("Unexpected closing parenthesis"),
+        Some(quotting) => {
+            let (sym_quote, start) = match quotting {
+                Token::Quote(s) => (Sym::new("quote"), s),
+                Token::QuasiQuote(s) => (Sym::new("quasiquote"), s),
+                Token::Unquote(s) => (Sym::new("unquote"), s),
+                _ => bail!("Unexpected token, should be Quote, QuasiQuote or Unquote"),
```

Comment:
Should be `unreachable!(...)` : if you reach this it is a bug in our code, not in the user input.